### PR TITLE
lint: ruff

### PR DESCRIPTION
**Pull request by elviskahoro**
lint

### CODE REVIEW

**Comment by elviskahoro** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/reflex-dev/reflex-web/887?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#887** <a href="https://app.graphite.dev/github/pr/reflex-dev/reflex-web/887?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#886** <a href="https://app.graphite.dev/github/pr/reflex-dev/reflex-web/886?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @elviskahoro and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by picklelo — APPROVED** [review]

File: docs/tutorial/tutorial_utils.py [hunk 1]
```diff
@@ -13,12 +13,12 @@ class ChatappState(rx.State):
     # Keep track of the chat history as a list of (question, answer) tuples.
     chat_history: list[tuple[str, str]]
 
-    def answer(self):
+    def answer(self) -> None:
```
Comment (picklelo) [inline docs/tutorial/tutorial_utils.py:16]: Is it possible to turn off -> None ? I don't think this is a requirement in black

File: pcweb/components/webpage/navbar/inkeep.py [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
-    special_props: Set[Var] = {
-        Var.create_safe("{...searchBarProps}", _var_is_string=False)
+    special_props: set[Var] = {
+        Var.create_safe("{...searchBarProps}", _var_is_string=False),
     }
 
     is_open: Var[bool] = False
 
-    on_close: EventHandler[lambda: []]
+    on_close: EventHandler[list]
```
Comment (picklelo) [inline pcweb/components/webpage/navbar/inkeep.py:18]: I think we want to keep the original way as it's the explicit way to define specs

File: pcweb/components/webpage/navbar/navbar.py [hunk 3]
```diff
@@ -41,39 +43,22 @@ def resources_item(text, url, icon):
 #     return rx.hover_card.root(
 #         rx.hover_card.trigger(
 #             rx.flex(
-#                 rx.text("Resources", style=style),
```
Comment (picklelo) [inline pcweb/components/webpage/navbar/navbar.py:44]: why does this get deleted?

File: pcweb/flexdown.py [hunk 4]
```diff
@@ -1,25 +1,23 @@
...
-    text_comp,
-    list_comp,
-    definition,
-)
 
-from pcweb.styles.fonts import base, code
+from pcweb.styles.colors import c_color
+from pcweb.styles.fonts import base
+from pcweb.styles.fonts import code
+from pcweb.templates.docpage import code_block_markdown
```
Comment (picklelo) [inline pcweb/flexdown.py:7]: Can we keep the other style imports, this is quite verbose

**Review by picklelo — CHANGES_REQUESTED** [review]
Nice to get this listed - some comments on if we can configure the linter to match what we have in reflex, I like the listing rules we have on that repo.

**Comment by elviskahoro** [comment]
@picklelo gonna close this PR and then eventually re-open a new one that goes through some of the stuff you've mentioned in these comments.

Here's the ruff.toml config I have setup. It's a superset of reflex core's setup -> it has all of the rules from reflex core but it's more strict (ruff default is a pretty low bar). I've gone through and added ignores for the sub-rules of the new strict ones so that the messages are still relevant to our repo!

File: pcweb/components/webpage/navbar/inkeep.py (hunk 2 above)
Comment (elviskahoro) [reply pcweb/components/webpage/navbar/inkeep.py:18]: Fixed! In the new PR that i'll open!

File: pcweb/flexdown.py (hunk 4 above)
Comment (elviskahoro) [reply pcweb/flexdown.py:7]: Sure, we can keep them together line.

**Pull request closed without merging**